web/src/lib/utility.ts

```diff
@@ -47,17 +47,35 @@ export function isFocus(): boolean {
 }
 
 /**
+ * Test whether this is a browser on iOS.
+ */
+export function isIOS(): boolean {
+  return /(iPod|iPhone|iPad)/i.test(window.navigator.userAgent);
+}
+
+/**
+ * Check whether the browser is Safari (either desktop or mobile).
+ */
+export function isSafari(): boolean {
+  const userAgent = window.navigator.userAgent;
+  /* Just checking isSafari isn't enough, because multiple browsers on iOS
+   * identify as Safari. The difference is that they have a different version
+   * string in the user agent. E.g. Safari has Version/<version>, Chrome has
+   * CriOS/<version>, Firefox has FxiOS/<version>.
+   */
+  const isWebkit = /AppleWebKit/i.test(userAgent);
+  const pretendsSafari = /Safari/i.test(userAgent);
+  const isSafari = /Version/i.test(userAgent);
+  return isWebkit && pretendsSafari && isSafari;
+}
+
+/**
  * Check whether the browser is mobile Safari (i.e. on iOS).
- * 
+ *
  * The logic is collected from answers to this SO question: https://stackoverflow.com/q/3007480
  */
 export function isMobileSafari(): boolean {
```

Comment:
This changes the meaning of this function slightly. Hindsight 20/20, this function should be called `isMobileWebkit.` We also use it to make buttons behave better on iOS browsers (thanks @nevik !). We still want that functionality for FF and Chrome on iOS, but this disables it. Probably best to leave this function mostly untouched, except for maybe swapping the `isIOS` function with the `isWebkit` function, since we would use the latter in two places.